storage/temporal_storage.go

```diff
@@ -0,0 +1,374 @@
+	// Maximum number of idle connections in the pool.
+	MaxIdle int `json:"optimisation_max_idle" mapstructure:"optimisation_max_idle"`
+	// Maximum number of connections allocated by the pool at a given time. When zero, there is no
+	// limit on the number of connections in the pool. Defaults to 500.
+	MaxActive int `json:"optimisation_max_active" mapstructure:"optimisation_max_active"`
+
+	// Enable this option if you are using a cluster instance. Default is `false`.
+	EnableCluster bool `json:"enable_cluster" mapstructure:"enable_cluster"`
+	// Setting this to true to use SSL when connecting to the DB.
+	// Deprecated: use UseSSL instead.
+	RedisUseSSL bool `json:"redis_use_ssl" mapstructure:"redis_use_ssl"`
+	// Set this to `true` to tell Pump to ignore database's cert validation.
+	// Deprecated: use SSLInsecureSkipVerify instead.
+	RedisSSLInsecureSkipVerify bool `json:"redis_ssl_insecure_skip_verify" mapstructure:"redis_ssl_insecure_skip_verify"`
+	// Setting this to true to use SSL when connecting to the DB.
+	UseSSL bool `json:"use_ssl" mapstructure:"use_ssl"`
+	// Set this to `true` to tell Pump to ignore database's cert validation.
+	SSLInsecureSkipVerify bool `json:"ssl_insecure_skip_verify" mapstructure:"ssl_insecure_skip_verify"`
+}
+
+// TemporalStorageHandler is a storage manager that uses non data-persistent databases, like Redis.
+type TemporalStorageHandler struct {
+	KeyPrefix string
+	db        *storageHandler
+	Config    TemporalStorageConfig
+	HashKeys  bool
+}
+
+func NewTemporalStorageHandler(forceReconnect bool, config *TemporalStorageConfig) error {
+	switch config.Type {
+	case "redis", "":
+		if !forceReconnect {
+			if temporalStorageSingleton != nil {
+				log.WithFields(logrus.Fields{
+					"prefix": logPrefix,
+				}).Debug("Redis pool already INITIALISED")
+				return nil
+			}
+		} else {
+			if temporalStorageSingleton != nil {
+				err := temporalStorageSingleton.conn.Disconnect(ctx)
+				if err != nil {
+					return fmt.Errorf("error disconnecting Redis: %s", err.Error())
+				}
+			}
+		}
+
+		log.WithFields(logrus.Fields{
+			"prefix": logPrefix,
+		}).Debug("Creating new Redis connection pool")
+
+		maxActive := 500
+		if config.MaxActive > 0 {
+			maxActive = config.MaxActive
+		}
+
+		timeout := 5
+
+		if config.Timeout > 0 {
+			timeout = config.Timeout
+		}
+
+		opts := &model.RedisOptions{
+			MasterName:       config.MasterName,
+			SentinelPassword: config.SentinelPassword,
+			Addrs:            config.Addrs,
+			Database:         config.Database,
+			Username:         config.Username,
+			Password:         config.Password,
+			MaxActive:        maxActive,
+			Timeout:          timeout,
+			EnableCluster:    config.EnableCluster,
+			Host:             config.Host,
+			Port:             config.Port,
+			Hosts:            config.Hosts,
+		}
+
+		enableTLS := config.UseSSL
+		if !enableTLS {
+			enableTLS = config.RedisUseSSL
+		}
+
+		insecureSkipVerify := config.SSLInsecureSkipVerify
```

Comment:
this can be simplified to `insecureSkipVerify := config.SSLInsecureSkipVerify ||  config.RedisSSLInsecureSkipVerify`